lms/static/js/student_account/views/FinishAuthView.js

```diff
@@ -152,13 +155,36 @@
                 }
             },
 
+            /**
+             * Check if the target url is same as current domain or a subdomain.
+             * @param url Target url
+             * @returns {boolean}
+             */
+            isAllowedOrigin: function( url ) {
+                url = !url.includes('http') ? location.protocol + '//' + url : url;
+                var target = new URL(url);
+                var isSameOrSubDomain = (target.host.includes(location.host.replace('www.', '')) || location.host.includes(target.host.replace('www.', '')))
```

Comment:
Break this down. I think this will fail in many instances. For example, if progs is talking to edX, this will definitely fail. It only addresses marketing. I believe you should use the domains and subdomain names as set in settings.

Reply:
My mistake. Good catch 👍